Mathlib/Computability/DFA.lean

```diff
@@ -164,6 +164,116 @@ theorem pumping_lemma [Fintype σ] {x : List α} (hx : x ∈ M.accepts)
   have h := M.evalFrom_of_pow hb hb'
   rwa [mem_accepts, eval, evalFrom_of_append, evalFrom_of_append, h, hc]
 
+section SetClosure
+
+variable {σ' : Type v}
+
+/--
+ `M.compl` constructs a DFA for the complement of the language of `M`.
+
+ Use `Mᶜ` rather than directly using this function.
+ -/
+def compl (M : DFA α σ) : DFA α σ where
+  step := M.step
+  start := M.start
+  accept := M.acceptᶜ
+
+instance : HasCompl (DFA α σ) := ⟨compl⟩
+
+@[simp]
+theorem compl_accept_eq_accept_compl : Mᶜ.accept = M.acceptᶜ := rfl
+
+@[simp]
+theorem compl_accepts_eq_accepts_compl : Mᶜ.accepts = M.acceptsᶜ := rfl
+
+theorem compl_accept_iff (s : σ) : s ∈ Mᶜ.accept ↔ s ∉ M.accept := by simp
```

Comment:
```suggestion
theorem mem_accept_compl {s : σ} : s ∈ Mᶜ.accept ↔ s ∉ M.accept := by simp
```
Same below

Reply:
I made these changes for the theorems about complements. For other set operations (union and intersection), what would be good choices for making parameters implicit w.r.t. mathlib4 expectations?

For example, currently I have:

```lean
theorem inter_accept_iff (M₁ : DFA α σ) (M₂ : DFA α σ') (s : σ × σ') :
    s ∈ (M₁.inter M₂).accept ↔ s.fst ∈ M₁.accept ∧ s.snd ∈ M₂.accept := by
```

I will rename it to `mem_accept_inter`. My question is should I:
- Make `s` implicit?
- Make all three of `M₁`, `M₂`, `s` implicit?

I think the first option makes sense (it'd mirror the signature of `mem_accept_compl`).

Reply:
Iffs should in general have as many implicit arguments as possible so that one can write `iff_lemma.1`/`iff_lemma.2` to get access to the forward and backward implications